### PR TITLE
Fix serializing error with fatal property.

### PR DESCRIPTION
**Pull request by erychagov**
Added test for this case. Also I sorted files and folders by type and name under MobileCenterCrashesTests folder

### CODE REVIEW

File: MobileCenterCrashes/MobileCenterCrashesTests/MSCrashesTests.mm [hunk 1]
```diff
@@ -505,6 +506,28 @@ - (void)testWrapperCrashCallback {
...
+  MSAbstractErrorLog *log = [MSAbstractErrorLog new];
+
+  // If
+  log.fatal = NO;
+
+  // When
+  NSDictionary *serializedLog = [log serializeToDictionary];
+
+  // Then
+  XCTAssertNotNil([serializedLog objectForKey:kMSFatal]);
```
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashesTests/MSCrashesTests.mm]: Check the value is NO.

File: MobileCenterCrashes/MobileCenterCrashesTests/MSCrashesTests.mm [hunk 2]
```diff
@@ -505,6 +506,28 @@ - (void)testWrapperCrashCallback {
...
+
+  // If
+  log.fatal = YES;
+
+  // When
+  serializedLog = [log serializeToDictionary];
+
+  // Then
+  XCTAssertNotNil([serializedLog objectForKey:kMSFatal]);
```
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashesTests/MSCrashesTests.mm]: Check the value is YES.

File: MobileCenterCrashes/MobileCenterCrashesTests/MSCrashesTests.mm [hunk 3]
```diff
@@ -505,6 +506,28 @@ - (void)testWrapperCrashCallback {
   XCTAssertTrue(first.count == 1);
 }
 
+- (void)testAbstractErrorLogSerialization {
+  MSAbstractErrorLog *log = [MSAbstractErrorLog new];
```
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashesTests/MSCrashesTests.mm]: You probably want to test the value is NO after serialization even though you don't assign any value to log.fatal.

**Comment by dhei** [comment]
Looks good to me

**Pull request merged**